sys/kern/imgact_elf.c

```diff
@@ -789,139 +809,16 @@ __elfN(load_section)(const struct image_params *imgp, vm_ooffset_t offset,
-	 * For RTLD we also align upwards to avoid aligning down into the
-	 * memory region for the main binary.
-	 */
-	adjusted_base = roundup2(*preferred_rbase,
-	    CHERI_REPRESENTABLE_ALIGNMENT(total_size));
-	uprintf("Attempted to load %s with non-representable mapping: moving "
-	    "from %#jx to %#jx\n", execpath, (uintmax_t)*preferred_rbase,
-	    (uintmax_t)adjusted_base);
-	*preferred_rbase = adjusted_base;
-	return (0);
-}
-
-/*
- * Claim the space around the executable that is required to represent a
- * capability spanning the executable.
- */
-static int
-__elfN(reserve_representability_padding)(const char *execpath, vm_map_t map,
-    Elf_Addr mapbase, Elf_Addr start_addr, Elf_Addr end_addr,
-    u_long *representable_start, u_long *representable_end)
-{
-	size_t representable_length;
-	int error;
-	size_t maplen = end_addr - start_addr;
-	/*
-	 * Sanity check that the base address was aligned correctly so that we
-	 * can represent a capability spanning the entire executable.
-	 */
-	MPASS2(trunc_page(mapbase) == mapbase, "Not page-aligned?");
-	MPASS2(trunc_page(start_addr) == start_addr, "Not page-aligned?");
-	*representable_start = CHERI_REPRESENTABLE_BASE(start_addr, maplen);
-	MPASS(mapbase <= *representable_start);
-	representable_length = CHERI_REPRESENTABLE_LENGTH(maplen);
-	if (*representable_start != start_addr) {
-		/*
-		 * This should rarely be needed, so keep an uprintf() to
-		 * discover weird ELF files.
-		 */
-		uprintf("%s: Reserving space before start: [%#jx-%#jx] for "
-		    "%s mapping at [%#jx-%#jx]\n", __func__,
-		    (uintmax_t)*representable_start, (uintmax_t)start_addr,
-		    execpath, (uintmax_t)start_addr, (uintmax_t)end_addr);
-		error = reserve_address_space(map, *representable_start,
-		    start_addr, /*reservation=*/start_addr, execpath);
-		if (error)
-			return (error);
-	}
-	*representable_end = *representable_start +
-	    CHERI_REPRESENTABLE_LENGTH(maplen);
-	/* We must have mapped up to the end of a page. */
-	if (round_page(*representable_end) != round_page(end_addr)) {
-		/*
-		 * This should rarely be needed, so keep an uprintf() to
-		 * discover weird ELF files.
-		 */
-		uprintf("%s: Reserving space after end: [%#jx-%#jx] for "
-		    "%s mapping at [%#jx-%#jx]\n", __func__,
-		    (uintmax_t)round_page(end_addr),
-		    (uintmax_t)round_page(*representable_end), execpath,
-		    (uintmax_t)start_addr, (uintmax_t)end_addr);
-		error = reserve_address_space(map, round_page(end_addr),
-		    round_page(*representable_end), /*reservation=*/start_addr,
-		    execpath);
-		if (error)
-			return (error);
-	}
-	return (0);
-}
-#endif
-
 static int
 __elfN(load_sections)(const struct image_params *imgp, const Elf_Ehdr *hdr,
     const Elf_Phdr *phdr, u_long rbase, u_long *base_addrp, u_long *max_addrp,
-    u_long *representable_start, u_long *representable_end, const char *file)
+    u_long *representable_start, u_long *representable_end)
```

Comment:
@arichardson When rebasing I left in the `representable_start` and `representable_end` as it makes setting `imgp->star_addr` and `imgp->end_addr` and equivalent `interp_` versions. I wonder whether we can drop them and use the `base_addrp` and `max_addrp` arguments, do you have an opinion on this?

Reply:
We can probably drop them since the reservation logic ensure representable bounds, so we can just use the macros to get the representable values.